### PR TITLE
Setup local storage

### PR DESCRIPTION
**Pull request by MindPrepDev**
This branch adds local storage functionality to the portfolio site to enhance accessibility and improve user experience.
How it achieves this feat is by:

- Preventing default form behaviour i.e submission before capturing user's input
- Checking for any past interactivity between the user and the platform and dynamically upload the inputs found
- If not found, it then goes on to set new inputs into the storage
- Lastly, it submits the form

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -385,19 +385,45 @@ document.addEventListener('DOMContentLoaded', () => {
...
+    email.value = getInfo.user_email;
+    message.value = getInfo.user_message;
+    if (!localStorage.getItem('userInfo')) {
+      userData = {
+        user_name: userName.value,
+        user_email: email.value,
+        user_message: message.value,
+      };
+      localStorage.setItem('userInfo', JSON.stringify(userData));
+    }
```
Comment (roshan-bajgain) [inline main.js:422]: - [ ] After filling out the form, and not submitting it, the entered values should be saved in the local storage. kindly implement it as it is not saving
### Image
ss

File: main.js [hunk 2]
```diff
@@ -385,19 +385,45 @@ document.addEventListener('DOMContentLoaded', () => {
...
+    userName.value = getInfo.user_name;
+    email.value = getInfo.user_email;
+    message.value = getInfo.user_message;
+    if (!localStorage.getItem('userInfo')) {
+      userData = {
+        user_name: userName.value,
+        user_email: email.value,
+        user_message: message.value,
+      };
+      localStorage.setItem('userInfo', JSON.stringify(userData));
```
Comment (roshan-bajgain) [inline main.js:421]: - [ ] When the user loads the page if there is any data in the local storage the input fields are pre-filled with this data. kindly implement it as well

**Review by roshan-bajgain — CHANGES_REQUESTED** [review]
Hi @MindPrepDev, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- No linter errors. ✔️
- GitHub flow is followed. ✔️
- Professional README file. ✔️
- PR has a good title and summary. ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @MindPrepDev @roshan-bajgain 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**